test-apps/presentation-test-app/src/frontend/api/MyAppFrontend.ts

```diff
@@ -19,13 +29,11 @@ export class MyAppFrontend {
     return SampleRpcInterface.getClient().getAvailableRulesets();
   }
 
-  public static async openIModel(path: string): Promise<IModelConnection> {
+  public static async openIModel(path: string): Promise<IModelConnection | undefined> {
     if (IpcApp.isValid) {
       Logger.logInfo("presentation", `Trying to open standalone ${path}`);
       this.iModel = await tryOpenStandalone(path);
-    }
-
-    if (!this.iModel) {
+    } else {
```

Comment:
Open standalone might fail if `StandaloneEdit` is not enabled on imodel. In that case I think we want to fallback on opening it as a snapshot if we want to do something with non editable imodel in electron app?

Reply:
Fixed, see #1290